test/sqllogictest/transform/scalar_cse.slt

```diff
@@ -0,0 +1,64 @@
+# Copyright Materialize, Inc. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# This file contains tests for the Scalar CSE transforms.
+
```

Comment:
So far, I've just skimmed the implementation, but do I have this right:
This PR doesn't support commutative functions yet, like say,
`select a*b*2, b*a+1 from x`.
and it might not support associative functions.

Reply:
This is definitely correct. There are lots of identities that are available to us and we might want to think about how to use them. I think there are some pretty complicated holes we could go down, but we could start with a normalized representation of expressions (e.g. perhaps always sort commutative functions so that the first argument is the "lesser").